### PR TITLE
Fix and document Vertex AI orchestrator and step operator

### PR DESCRIPTION
**Pull request by stefannica**
## Describe changes
This PR fixes the Vertex AI orchestrator scheduled pipelines feature to work with the GCP service connector. It also updates the Vertex AI orchestrator and step operator documentation to cover the new ways of authenticating and granting permissions to GCP resources.

## Pre-requisites
Please ensure you have done the following:
- [x] I have read the **CONTRIBUTING.md** document.
- [x] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] If I have added an integration, I have updated the [integrations](https://docs.zenml.io/component-gallery/integrations) table and the [corresponding website section](https://zenml.io/integrations).
- [ ] I have added tests to cover my changes.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other (add details above)

### CODE REVIEW

**Review by AlexejPenner — APPROVED** [review]
:seal:

**Review by schustmi — APPROVED** [review]

File: docs/book/user-guide/component-guide/orchestrators/vertex.md [hunk 1]
```diff
@@ -50,24 +50,169 @@ To use the Vertex orchestrator, we need:
...
+
+    * permissions to create a job in Vertex Pipelines, (e.g. [the `Vertex AI User` role](https://cloud.google.com/vertex-ai/docs/general/access-control#aiplatform.user)).
+
+4. *the Google Cloud Scheduler* is a GCP resource that is used to trigger the pipeline on a schedule. This component is only needed if you intend on running Vertex AI pipelines on a schedule. The scheduler needs a GCP service account to authenticate to *the scheduler Google Cloud Function*. Let's call this service account *the scheduler service account*. *The scheduler service account* can be explicitly configured in the orchestrator configuration via the `scheduler_service_account` parameter. If it is omitted, the orchestrator will use the following, in order of precedence:
```
Comment (bcdurak) [inline docs/book/user-guide/component-guide/orchestrators/vertex.md:94]: In points 3 and 4, I would explain their responsibilities a bit more in detail if possible. As far as I know, in the case of a scheduled pipeline, the cloud function takes over the responsibility of launching the vertex pipeline, whereas the scheduler makes sure that the function is executed on the given schedule.

File: docs/book/user-guide/component-guide/orchestrators/vertex.md [hunk 2]
```diff
@@ -50,24 +50,169 @@ To use the Vertex orchestrator, we need:
...
+
+#### Configuration use-case: local `gcloud` CLI with user account
+
+This configuration use-case assumes you have configured the [`gcloud` CLI](https://cloud.google.com/sdk/gcloud) to authenticate locally with your GCP account (i.e. by running `gcloud auth login`). It also assumes the following:
+
+* you are not planning to run pipelines on a schedule.
+* your GCP account has permissions to create a job in Vertex Pipelines, (e.g. [the `Vertex AI User` role](https://cloud.google.com/vertex-ai/docs/general/access-control#aiplatform.user)).
+* [the Compute Engine default service account](https://cloud.google.com/compute/docs/access/service-accounts#default_service_account) for the GCP project in which the pipeline is running is updated with additional permissions required to run a Vertex AI pipeline, (e.g. [the `Vertex AI Service Agent` role](https://cloud.google.com/vertex-ai/docs/general/access-control#aiplatform.serviceAgent)).
+
+{% hint style="info" %}
```
Comment (bcdurak) [inline docs/book/user-guide/component-guide/orchestrators/vertex.md:115]: I would change this into a warning.
```suggestion
{% hint style="warning" %}
```

File: docs/book/user-guide/component-guide/orchestrators/vertex.md [hunk 3]
```diff
@@ -50,24 +50,169 @@ To use the Vertex orchestrator, we need:
...
+
+You also have three different options to provide credentials to the orchestrator:
+
+* use the [`gcloud` CLI](https://cloud.google.com/sdk/gcloud) to authenticate locally with GCP
+* configure the orchestrator to use a [service account key file](https://cloud.google.com/iam/docs/creating-managing-service-account-keys) to authenticate with GCP by setting the `service_account_path` parameter in the orchestrator configuration.
+* (recommended) configure [a GCP Service Connector](../../../platform-guide/set-up-your-mlops-platform/connect-zenml-to-infrastructure/gcp-service-connector.md) with GCP credentials and then link the Vertex AI Orchestrator stack component to the Service Connector.
+
+This section [explains the different components and GCP resources](#vertex-ai-pipeline-components) involved in running a Vertex AI pipeline and what permissions they need, then provides instructions for three different configuration use-cases:
```
Comment (bcdurak) [inline docs/book/user-guide/component-guide/orchestrators/vertex.md:67]:
```suggestion
[The next section](#vertex-ai-pipeline-components) explains the different components and GCP resources involved in running a Vertex AI pipeline and what permissions they need, then provides instructions for three different configuration use-cases:
```

**Pull request merged**